### ISSUE TITLE
Add support for gathering storage device information from node annotations and labels

### ISSUE THREAD

**Issue by solidDoWant** [rook/rook#15010]
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**

Allow for pulling storage device paths/names from node annotations or labels.

**What is use case behind this feature:**

Determining what device(s) can be used by Ceph is often done prior to deploying a node (physical or virtual, on-prem or in cloud). Node labels and annotations are a great way to inform workloads about a node's capabilities, such as what hardware the node is using, or what "failure domain" it's in.

If Rook supported pulling drive names/paths/identifiers or other hardware-specific storage configuration from nodes, this would allow Rook to learn node-specific deployment information without needing to update the cluster configuration every time a node is added. This could also be useful for when separate groups are managing the underlying infrastructure (OS downwards) and platform (Kubernetes/k8s workloads upwards). The group that actual deploys the hardware resources would be able to programmatically provide hardware information (what drives are available) to workloads (Rook-Ceph). In many cases, this would allow for a single source of truth, where the infrastructure group provides a list of drives and labels/annotations to automation tools (Terraform, Ansible, etc.).

**Environment**:

Any environment utilizing non-PVC disks for Rook storage

**Comment by parth-gr** [comment]
@solidDoWant Yes we may use node annotations to decide the deployments, Can you shed some more light on the benefit of this feature?
cc @travisn

**Comment by travisn** [comment]
There have been discussions about this in the past, perhaps in slack or at conferences. In particular, this approach would likely supersede the [nodes](https://github.com/rook/rook/blob/master/deploy/examples/cluster.yaml#L278-L286) schema that is currently in the CephCluster CR. To keep the implementation simple, perhaps we could even use the same node schema and serialize it to json for the node annotation.
